lib/build.ts

```diff
@@ -328,32 +333,21 @@ async function checkAllowDockerPlatformHandling(
 		return true;
 	} else if (imagesWithPlatformSupport.length > 0) {
 		// Mixed images
-		console.log(
-			'Warning: Dockerfile contains a mixes images that require a platform selection with ones that do not support it.\n' +
-				'The following images do not support platform selection:\n-' +
-				imagesWithoutPlatformSupport.join('\n-') +
-				'\n\n' +
-				'The following images require platform selection:\n-' +
-				imagesWithPlatformSupport.join('\n-') +
-				'\n\n' +
-				'To avoid a possible build error, the CLI has disabled platform selection. ' +
-				'As a result, the architecture of the machine where the Docker Engine ' +
-				'is running will be used to select the platform when pulling upstream images ' +
-				'which may result in "exec format error" at runtime. To avoid this warning, ' +
-				'update or replace the images that do not support platform selection.',
-		);
-	} else {
-		// Only images without platform support
-		console.log(
-			'Warning: Dockerfile contains a images do not support platform selection.\n-' +
-				imagesWithoutPlatformSupport.join('\n-') +
-				'\n\n' +
-				'To avoid a possible build error, the CLI has disabled platform selection. ' +
-				'As a result, the architecture of the machine where the Docker Engine ' +
-				'is running will be used to select the platform when pulling upstream images ' +
-				'which may result in "exec format error" at runtime. To avoid this warning, ' +
-				'update or replace the images that do not support platform selection.',
-		);
+		warn(`
+Service '${task.serviceName}': A multistage Dockerfile contains a mix of
+base images that require CPU architecture selection and base images that
+do not support it.
+The following base images do not support CPU architecture selection:
+-${imagesWithoutPlatformSupport.join('\n-')}
+
+The following base images require CPU architecture selection:
+-${imagesWithPlatformSupport.join('\n-')}
+
+As a result, the CPU architecture of the machine where the Docker Engine
+is running will be used to select base images that require architecture
+selection. This may result in incorrect architecture selection and "exec
+format error" at runtime. To avoid this warning, update or replace base
+images that do not support architecture selection.`);
```

Comment:
I haven't tested this yet, not sure it looks good. I will test it with balena-io/balena-cli/pull/2301 .